src/transaction/flashback.c

```diff
@@ -55,6 +55,177 @@
 #include "thread_entry.hpp"
 #include "oid.h"
 #include "storage_common.h"
+#include "system_parameter.h"
+
+static volatile LOG_PAGEID flashback_Min_log_pageid = NULL_LOG_PAGEID;	// Minumun log pageid to keep archive log volume from being removed
+static volatile time_t flashback_Last_request_done_time = -1;	// The time most recently requested by flashback
+static volatile int flashback_Threshold_to_remove_archive = 0;	/* If the difference between the time at which the archive log is deleted
+								 * and the time the flashback last requested exceeds this threshold,
+								 * the archive log volume can be deleted.
+								 */
+// *INDENT-OFF*
+static std::atomic_bool flashback_Is_active = false;	// the status value that the flashback is processing the request
+// *INDENT-ON*
+
+/*
+ * flashback_set_min_log_pageid_to_keep - set flashback_Min_log_pageid
+ */
+
+void
+flashback_set_min_log_pageid_to_keep (LOG_LSA * lsa)
+{
+  assert (lsa != NULL);
+
+  flashback_Min_log_pageid = lsa->pageid;
+}
+
+/*
+ * flashback_unset_min_log_pageid_to_keep - unset flashback_Min_log_pageid to NULL_LOG_PAGEID
+ */
+
+void
+flashback_unset_min_log_pageid_to_keep ()
+{
+  flashback_Min_log_pageid = NULL_LOG_PAGEID;
+}
+
+/*
+ * flashback_min_log_pageid_to_keep - returns minimum log pageid to keep
+ *
+ * return   : minimum pageid that flashback have to keep
+ */
+
+LOG_PAGEID
+flashback_min_log_pageid_to_keep ()
+{
+  return flashback_Min_log_pageid;
+}
+
+/*
+ * flashback_set_request_done_time - set flashback_Last_request_done_time to current
+ */
+
+void
+flashback_set_request_done_time ()
+{
+  flashback_Last_request_done_time = time (NULL);
+}
+
+/*
+ * flashback_set_threshold - set flashback_Threshold_to_remove_archive
+ *                           it follows PRM_ID_REMOVE_LOG_ARCHIVES_INTERVAL
+ */
+
+static void
+flashback_set_threshold ()
+{
+  flashback_Threshold_to_remove_archive = prm_get_integer_value (PRM_ID_REMOVE_LOG_ARCHIVES_INTERVAL);
+  if (flashback_Threshold_to_remove_archive < 60)
+    {
+      /* set threshold to 60 which is recommended intervals for removing archive logs */
+      flashback_Threshold_to_remove_archive = 60;
+    }
+}
+
+/*
+ * flashback_check_time_to_remove_archive - check the time if the archive can be removed
+ *
+ * return   : true or false
+ */
+
+bool
+flashback_check_time_exceed_threshold ()
+{
+  if (flashback_Threshold_to_remove_archive == 0)
```

Comment:
`flashback_Threshold_to_remove_archive` is used only in this function. Do we really need to declare this as a global variable? It seems that one local variable can be used for it like:
```
int threshold = prm_get_integer_value (PRM_ID_REMOVE_LOG_ARCHIVES_INTERVAL);
threshold = threshold < 60 ? 60 : threshold.
```

Reply:
`flashback_set_threshold` can be also removed.

Reply:
If that is your concern, you can use a static local variable, but I think it doesn't matter.